core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/ApplyUserAgentStage.java

```diff
@@ -140,6 +154,32 @@ private String getUserAgent(SdkClientConfiguration config, List<ApiName> request
         return userAgent.toString();
     }
 
+    private static Optional<String> identityProviderName(ExecutionAttributes executionAttributes) {
+        SelectedAuthScheme<?> selectedAuthScheme = executionAttributes
+            .getAttribute(SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME);
+        if (selectedAuthScheme == null) {
+            return Optional.empty();
+        }
+        return providerNameFromIdentity(selectedAuthScheme);
+    }
+
+    private static <T extends Identity> Optional<String> providerNameFromIdentity(SelectedAuthScheme<T> selectedAuthScheme) {
+        CompletableFuture<? extends T> identityFuture = selectedAuthScheme.identity();
+        T identity = CompletableFutureUtils.joinLikeSync(identityFuture);
```

Comment:
Looks like this is another place where we are making it wait for a future. Do we just want to add a comment indicating that we are blocking here and we eventually would need to remove this ?

Reply:
I don't think it's necessary; it's easy to search for this in the code and that's something we'd have to do anyway when we've unblocked async identity handling.